app/assets/javascripts/discourse/tests/integration/components/user-menu/messages-list-test.js

```diff
@@ -25,6 +139,113 @@ module("Integration | Component | user-menu | messages-list", function (hooks) {
+      copy.read_notifications = [getGroupMessageSummaryNotification()];
+      return response(copy);
+    });
+
+    await render(template);
+    const items = queryAll("ul li");
+
+    assert.strictEqual(items.length, 2);
+
+    assert.ok(items[0].classList.contains("notification"));
+    assert.ok(items[0].classList.contains("unread"));
+    assert.ok(items[0].classList.contains("private-message"));
+
+    assert.ok(items[1].classList.contains("notification"));
+    assert.ok(items[1].classList.contains("read"));
+    assert.ok(items[1].classList.contains("group-message-summary"));
+  });
+
+  test("merge-sorts group_message_summary notifications and messages", async function (assert) {
+    pretender.get("/u/eviltrout/user-menu-private-messages", () => {
+      const copy = cloneJSON(
+        UserMenuFixtures["/u/:username/user-menu-private-messages"]
+      );
+      copy.unread_notifications = [];
+      copy.topics = [
+        getMessage({
+          bumped_at: "2014-07-26T01:29:24.177Z",
+          fancy_title: "Test Topic 0003",
+        }),
+        getMessage({
+          bumped_at: "2012-07-26T01:29:24.177Z",
+          fancy_title: "Test Topic 0002",
+        }),
+        getMessage({
+          bumped_at: "2010-07-26T01:29:24.177Z",
+          fancy_title: "Test Topic 0001",
+        }),
+      ];
+      copy.read_notifications = [
+        getGroupMessageSummaryNotification({
+          created_at: "2015-07-26T01:29:24.177Z",
+          data: {
+            inbox_count: 13,
+          },
+        }),
+        getGroupMessageSummaryNotification({
+          created_at: "2013-07-26T01:29:24.177Z",
+          data: {
+            inbox_count: 12,
+          },
+        }),
+        getGroupMessageSummaryNotification({
+          created_at: "2011-07-26T01:29:24.177Z",
+          data: {
+            inbox_count: 11,
+          },
+        }),
+      ];
+      return response(copy);
+    });
+    await render(template);
+    const items = queryAll("ul li");
+
+    assert.strictEqual(items.length, 6);
+
+    assert.strictEqual(
+      items[0].textContent.trim(),
+      I18n.t("notifications.group_message_summary", {
+        count: 13,
+        group_name: "jokers",
+      })
+    );
+    assert.strictEqual(
+      items[2].textContent.trim(),
+      I18n.t("notifications.group_message_summary", {
+        count: 12,
+        group_name: "jokers",
+      })
+    );
+    assert.strictEqual(
+      items[4].textContent.trim(),
+      I18n.t("notifications.group_message_summary", {
+        count: 11,
+        group_name: "jokers",
+      })
+    );
+
+    assert.strictEqual(
+      items[1].textContent.trim().replaceAll(/\s+/g, " "),
+      "mixtape Test Topic 0003"
+    );
+    assert.strictEqual(
+      items[3].textContent.trim().replaceAll(/\s+/g, " "),
+      "mixtape Test Topic 0002"
+    );
+    assert.strictEqual(
+      items[5].textContent.trim().replaceAll(/\s+/g, " "),
+      "mixtape Test Topic 0001"
+    );
```

Comment:
```suggestion
    assert.strictEqual(
      items[0].textContent.trim(),
      I18n.t("notifications.group_message_summary", {
        count: 13,
        group_name: "jokers",
      })
    );

    assert.strictEqual(
      items[1].textContent.trim().replaceAll(/\s+/g, " "),
      "mixtape Test Topic 0003"
    );

    assert.strictEqual(
      items[2].textContent.trim(),
      I18n.t("notifications.group_message_summary", {
        count: 12,
        group_name: "jokers",
      })
    );

    assert.strictEqual(
      items[3].textContent.trim().replaceAll(/\s+/g, " "),
      "mixtape Test Topic 0002"
    );

    assert.strictEqual(
      items[4].textContent.trim(),
      I18n.t("notifications.group_message_summary", {
        count: 11,
        group_name: "jokers",
      })
    );

    assert.strictEqual(
      items[5].textContent.trim().replaceAll(/\s+/g, " "),
      "mixtape Test Topic 0001"
    );
```

I feel like a sequential ordering is easier to read.